In [35]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import GridSearchCV
from sklearn.linear_model import LinearRegression
from sklearn.ensemble import RandomForestRegressor
import matplotlib.pyplot as plt
%matplotlib inline
import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)

In [6]:
data=pd.read_csv('housing_final_data.csv')

In [7]:
data.describe()

price         sqfeet           beds          baths  \
count  214381.000000  214381.000000  214381.000000  214381.000000   
mean     1072.366413     934.502684       1.817307       1.469529   
std       347.732507     238.719648       0.687699       0.543732   
min       265.000000     275.000000       1.000000       1.000000   
25%       800.000000     750.000000       1.000000       1.000000   
50%      1000.000000     924.000000       2.000000       1.000000   
75%      1310.000000    1100.000000       2.000000       2.000000   
max      1974.000000    1593.000000       3.000000       3.000000   

        cats_allowed   dogs_allowed  smoking_allowed  wheelchair_access  \
count  214381.000000  214381.000000    214381.000000      214381.000000   
mean        0.734753       0.711397         0.760739           0.069167   
std         0.441466       0.453114         0.426633           0.253738   
min         0.000000       0.000000         0.000000           0.000000   
25%         0.000000       0.000000         1.000000           0.000000   
50%         1.000000       1.000000         1.000000           0.000000   
75%         1.000000       1.000000         1.000000           0.000000   
max         1.000000       1.000000         1.000000           1.000000   

       electric_vehicle_charge    SF bay area  ...             mi  \
count            214381.000000  214381.000000  ...  214381.000000   
mean                  0.008970       0.001054  ...       0.059744   
std                   0.094285       0.032451  ...       0.237013   
min                   0.000000       0.000000  ...       0.000000   
25%                   0.000000       0.000000  ...       0.000000   
50%                   0.000000       0.000000  ...       0.000000   
75%                   0.000000       0.000000  ...       0.000000   
max                   1.000000       1.000000  ...       1.000000   

                  mn             ms             nc             nd  \
count  214381.000000  214381.000000  214381.000000  214381.000000   
mean        0.029233       0.020818       0.079737       0.014749   
std         0.168459       0.142775       0.270886       0.120549   
min         0.000000       0.000000       0.000000       0.000000   
25%         0.000000       0.000000       0.000000       0.000000   
50%         0.000000       0.000000       0.000000       0.000000   
75%         0.000000       0.000000       0.000000       0.000000   
max         1.000000       1.000000       1.000000       1.000000   

                  nj             nm             nv             ny  \
count  214381.000000  214381.000000  214381.000000  214381.000000   
mean        0.016895       0.010271       0.011195       0.033492   
std         0.128879       0.100826       0.105213       0.179917   
min         0.000000       0.000000       0.000000       0.000000   
25%         0.000000       0.000000       0.000000       0.000000   
50%         0.000000       0.000000       0.000000       0.000000   
75%         0.000000       0.000000       0.000000       0.000000   
max         1.000000       1.000000       1.000000       1.000000   

                  oh  
count  214381.000000  
mean        0.027139  
std         0.162488  
min         0.000000  
25%         0.000000  
50%         0.000000  
75%         0.000000  
max         1.000000  

[8 rows x 193 columns]

In [9]:
X=data.drop(['price'],axis=1)
y=data.price

In [10]:
X.head(2)

sqfeet  beds  baths  cats_allowed  dogs_allowed  smoking_allowed  \
0    1319     3      2             1             1                1   
1    1133     1      2             1             1                1   

   wheelchair_access  electric_vehicle_charge  SF bay area  akron / canton  \
0                  0                        0            0               0   
1                  0                        0            0               0   

   ...  mi  mn  ms  nc  nd  nj  nm  nv  ny  oh  
0  ...   0   0   0   0   0   0   0   0   0   0  
1  ...   0   0   0   0   0   0   0   0   0   0  

[2 rows x 192 columns]

In [11]:
y.head()

0    1120
1     825
2     800
3     785
4     900
Name: price, dtype: int64

In [13]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=101)

In [27]:
Scr=StandardScaler()
X1_train=Scr.fit_transform(X_train)
X1_test=Scr.fit_transform(X_test)
y1_train=y_train
y1_test=y_test

In [16]:
lrM=LinearRegression()
lrM.fit(X_train,y_train)

LinearRegression()

In [17]:
lrM.score(X_test,y_test)

0.6796354840119412

# Random Forest

In [31]:
RfM=RandomForestRegressor()

In [33]:
RfM.fit(X_train,y_train)

RandomForestRegressor()

In [34]:
RfM.score(X_test,y_test)

0.8726219169273715

## Conclusion

In this project using housing_Train DataSet to build  model which  predict Monthly housing rent given on following features. 
I used two different regressor(Linear Regression, Random Forest Regressor) to achive best score. 
The best score was achived using Random Forest Regressor

- Score 87%
